### ISSUE TITLE
Support Optional Chaining in Typescript Parser

### ISSUE THREAD

**Issue by justinmwarner** [typescript-eslint/typescript-eslint#1035]
**What code were you trying to parse?**

```ts
// Please put code here
  const obj = {name: ""};
  return {
    name: obj?.name,
    ...additionalData,
  };
```

**What did you expect to happen?**
No error, parses fine, and shows in IDE fine.

**What actually happened?**
![image](https://user-images.githubusercontent.com/2288818/66045423-55d44b80-e4e9-11e9-9225-92cef80328ef.png)
![image](https://user-images.githubusercontent.com/2288818/66045433-5bca2c80-e4e9-11e9-97f7-df40bb1132da.png)
![image](https://user-images.githubusercontent.com/2288818/66045440-61c00d80-e4e9-11e9-957f-d8da2c447037.png)

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `^2.3.1` |
| `TypeScript`                | `^3.7.0-dev.20191002` |
| `ESLint`                    | `6.1.0` |
| `node`                      | `10.16.3` |
| `npm`                       | `6.9.0` |

Expected to land in 3.7.x from Typescript-
https://github.com/Microsoft/TypeScript/wiki/Roadmap#37-november-2019
https://github.com/microsoft/TypeScript/pull/33294

Relates to #136

**Comment by AlCalzone** [comment]
This breaks Prettier auto-formatting too, if that is done via eslint.

**Comment by bradzacher** [comment]
3.7 beta was only just released (and the PRs for nullish coalescing / optional chaining were only merged a few days ago).

Please give us some time to add the features.

duplicate #1033

**Comment by AlCalzone** [comment]
No worries, just thought you might want to know the implications.

**Comment by matiishyn** [comment]
any updated on this one?

**Comment by justinmwarner** [comment]
> any updated on this one?

@matiishyn , look at #1033 , looks like eslint is good to go, prettier is still working on it. See the other ticket for the prettier items.
